flash/core/data/batch.py

```diff
@@ -251,6 +317,8 @@ def forward(self, batch: Sequence[Any]):
             else:
                 self.save_fn(final_preds)
         else:
+            if self.is_serving and isinstance(final_preds, list) and len(final_preds) == 1:
```

Comment:
Can you add an in-line comment on why this needs to be done?

Reply:
Done ! Added a todo.